File: leetcode3014/solution.cpp
class Solution {
public:
    int minimumPushes(string word) {
        int res=0, row=1, n=word.size();
        int quotient = n/8;
        int rem = n%8;

        while(quotient--) {
            // Add the product of 8 and the current row to the result
            res += 8 * row;

            // Increment the row for the next cycle
            row++;
        }
        // Product of remainder and final row accounts for the remaining
        // letters
        res += rem * row;
        return res;
    }
};

File: leetcode3014/solution.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        res=0
        n=len(word)
        row=1
        quotient = n//8
        rem=n%8
        while quotient > 0:
            # Add the product of 8 and the current row to the result
            res += 8 * row

            # increment the row for the next cycle and decrement the quotient
            quotient -= 1
            row += 1
        res += rem * row
        return res
